### PR TITLE
dnsdist: Deprecate syntactic sugar functions in 1.2.0

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Deprecate syntactic sugar functions in 1.2.0 for removal at a later time:
 * addAnyTCRule
 * addDomainBlock
 * addDomainSpoof
 * addDomainCNAMESpoof
 * addPoolRule
 * addNoRecurseRule
 * addDisableValidationRule
 * addQPSPoolRule
 * setDNSSECPool
 * addQPSLimit
 * addDelay

<strike>Don't merge yet, at the very least I need to update the new docs!</strike>

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
Updated to the new documentation format.

File: pdns/dnsdistdist/docs/rules-actions.rst [hunk 1]
```diff
@@ -84,26 +86,47 @@ Rule Generators
...
+  Deprecated as of 1.2.0, please use instead:
+
+    addAction(domain, DropAction())
 
   :param string domain: The domain name to block
 
 .. function:: addDomainSpoof(domain, IPv4[, IPv6])
               addDomainSpoof(domain, {IP[,...]})
 
+  .. deprecated:: 1.2.0
```
Comment (pieterlexis) [inline pdns/dnsdistdist/docs/rules-actions.rst]: All these deprecated things could have the line "Will be removed in 1.3.0" in there as well?

Comment (rgacogne) [inline pdns/dnsdistdist/docs/rules-actions.rst]: Yes, but I don't want to commit on doing the actual removal in 1.3.0 :)

Comment (rgacogne) [inline pdns/dnsdistdist/docs/rules-actions.rst]: Oh, I did in the warning messages when a rule is added, so it's not consistent. I guess I'll add the proposed line then!

**Pull request merged**